src/main/java/net/revelc/code/zookeeper/maven/plugin/StartZooKeeperMojo.java

```diff
@@ -37,7 +37,7 @@
 @Mojo(name = "start", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST, threadSafe = true)
 public class StartZooKeeperMojo extends AbstractZooKeeperMojo {
 
-  @Parameter(defaultValue = "${project.build.directory}/zmp", readonly = true)
+  @Parameter(defaultValue = "${project.build.directory}/zmp", property = "zmp.directory")
```

Comment:
To make this property optional, you should also set "alias" so that it can be configured in the plugin configuration in the POM, as well as via a property. For consistency, I've been setting `alias = "$x", property = "zmp.$x"`. I think `required = true` should also be there (but not certain, because maven developer documentation is poor) to prevent the default value from being overridden as empty (somehow). At the very least, for consistency with the other important fields in the generated docs.